cloudplatform/connectivity-destination-service/src/test/java/com/sap/cloud/sdk/cloudplatform/connectivity/DestinationServiceTest.java

```diff
@@ -584,6 +628,22 @@ void testGetAllDestinationsOnlySubscriberStrategyReadsSubscriberDestinations()
                 "The current tenant is the provider tenant, which should not be the case with the option OnlySubscriber. Cannot retrieve destination.");
     }
 
+    @Test
+    @DisplayName( "Test getting Destination Properties can enforce a subscriber tenant" )
+    void testGetAllDestinationPropertiesOnlySubscriberStrategyReadsSubscriberDestinations()
+    {
+        // set current tenant to be the provider tenant
+        context.clearTenant();
+
+        TenantAccessor
+            .executeWithTenant(
+                providerTenant,
+                () -> assertThatThrownBy(() -> loader.getAllDestinationProperties(ONLY_SUBSCRIBER))
+                    .isInstanceOf(DestinationAccessException.class)
+                    .hasMessageContaining(
+                        "The current tenant is the provider tenant, which should not be the case with the option OnlySubscriber. Cannot retrieve destination."));
```

Comment:
This seems redundant, the following should work:

```suggestion
        context.setTenant(providerTenant);
        assertThatThrownBy(() -> loader.getAllDestinationProperties(ONLY_SUBSCRIBER))
                    .isInstanceOf(DestinationAccessException.class)
                    .hasMessageContaining(
                        "The current tenant is the provider tenant, which should not be the case with the option OnlySubscriber. Cannot retrieve destination.");
```

Reply:
Works well!